src/vectors.jl

```diff
@@ -43,24 +43,18 @@ Vec3f(1, 2, 3) # explicitly ask for single precision
 """
 struct Vec{Dim,T} <: StaticVector{Dim,T}
   coords::NTuple{Dim,T}
-  Vec{Dim,T}(coords::NTuple{Dim,T}) where {Dim,T} = new{Dim,T}(coords)
-  Vec{Dim,T}(coords::NTuple{Dim,T}) where {Dim,T<:Integer} = new{Dim,Float64}(coords)
+  Vec(coords::NTuple{Dim,T}) where {Dim,T} = new{Dim,float(T)}(NTuple{Dim,float(T)}(coords))
```

Comment:
I think `NTuple{Dim,float(T)}` is not needed:
```suggestion
  Vec(coords::NTuple{Dim,T}) where {Dim,T} = new{Dim,float(T)}(coords)
```